apps/dashboard/src/app/(app)/team/[team_slug]/[project_slug]/(sidebar)/webhooks/analytics/components/WebhookAnalyticsCharts.tsx

```diff
@@ -0,0 +1,349 @@
+  }, [filteredLatencyData]);
+
+  // Chart configurations
+  const latencyChartConfig: ChartConfig = {
+    p50: {
+      color: "hsl(var(--chart-1))",
+      label: "P50 Latency",
+    },
+    p90: {
+      color: "hsl(var(--chart-2))",
+      label: "P90 Latency",
+    },
+    p99: {
+      color: "hsl(var(--chart-3))",
+      label: "P99 Latency",
+    },
+  };
+
+  // Generate status code chart config dynamically with class-based colors
+  const statusCodeConfig: ChartConfig = useMemo(() => {
+    const statusCodes = new Set<string>();
+    statusCodeData.forEach((item) => {
+      Object.keys(item).forEach((key) => {
+        if (key !== "time" && !Number.isNaN(Number.parseInt(key))) {
+          statusCodes.add(key);
+        }
+      });
+    });
+
+    const getColorForStatusCode = (statusCode: number): string => {
+      if (statusCode >= 200 && statusCode < 300) {
+        return "hsl(142, 76%, 36%)"; // Green for 2xx
+      } else if (statusCode >= 300 && statusCode < 400) {
+        return "hsl(48, 96%, 53%)"; // Yellow for 3xx
+      } else if (statusCode >= 400 && statusCode < 500) {
+        return "hsl(25, 95%, 53%)"; // Orange for 4xx
+      } else {
+        return "hsl(0, 84%, 60%)"; // Red for 5xx
+      }
+    };
+
+    const config: ChartConfig = {};
+    Array.from(statusCodes)
+      .sort((a, b) => {
+        const codeA = Number.parseInt(a);
+        const codeB = Number.parseInt(b);
+        return codeA - codeB;
+      })
+      .forEach((statusKey) => {
+        const statusCode = Number.parseInt(statusKey);
+        config[statusKey] = {
+          color: getColorForStatusCode(statusCode),
+          label: statusCode.toString(),
+        };
+      });
+
+    return config;
+  }, [statusCodeData]);
+
+  const selectedWebhookConfig = webhookConfigs.find(
+    (w) => w.id === selectedWebhook,
+  );
+
+  // Show empty state if no data
+  if (statusCodeData.length === 0 && latencyChartData.length === 0) {
+    return (
+      <div>
+        <div className="mb-4 flex justify-start">
+          <WebhookAnalyticsFilter />
+        </div>
+        <div className="flex items-center justify-center p-8 rounded-xl border border-border bg-card">
+          <div className="text-center">
+            <h3 className="text-lg font-semibold mb-2">
+              No webhook data available
+            </h3>
+            <p className="text-sm text-muted-foreground">
+              Webhook analytics will appear here once you start receiving
+              webhook events.
+            </p>
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <div className="mb-4 flex justify-start">
+        <WebhookAnalyticsFilter />
+      </div>
+      <ResponsiveSuspense
+        fallback={
+          <div className="flex flex-col gap-6">
+            <div className="flex flex-col gap-4 sm:flex-row sm:items-center sm:justify-between">
+              <div>
+                <Skeleton className="h-6 w-48" />
```

Comment:
Don't show skeletons for title, description and filters

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/yNOf1svJ8o3zjO7zQouZ/3fdf76d2-fb4a-459c-abd4-c14e7079c356.png)